### PR TITLE
Fix CmdLogMixin

### PR DESCRIPTION
**Pull request by aborilov**
we need to create ZapLogger manually because we don't want to use Sampler, that does not support arbitrary log levels

### CODE REVIEW

**Comment by nmiculinic** [comment]
Could you add some tests for this?

File: pkg/util/cmd.go [hunk 1]
```diff
@@ -36,11 +36,27 @@ func CmdLogMixin(cmd *cobra.Command) *cobra.Command {
...
+		if *dev {
+			ZapLogger = corezap.NewRaw(func(options *corezap.Options) {
+				level := zap.NewAtomicLevelAt(zapcore.Level(-*v))
+				options.Level = &level
+				options.Development = *dev
+			})
+		} else {
+			// we need to create ZapLogger manually because we don't want to use Sampler, that does not support arbitrary log levels
+			encoder := zapcore.NewJSONEncoder(zap.NewProductionEncoderConfig())
+			sink := zapcore.AddSync(os.Stderr)
```
Comment (nmiculinic) [inline pkg/util/cmd.go]:
```suggestion
			sink := zapcore.AddSync(cmd.ErrOrStderr())
```

this way it's easier to test if you want to inject error stream in the cobra command.

Comment (aborilov) [reply pkg/util/cmd.go]: thx, that was my first thought, how will I replace Stderr with my Writer

File: pkg/util/cmd.go [hunk 2]
```diff
@@ -30,19 +32,37 @@ import (
...
+		ZapLogger = corezap.NewRaw(func(options *corezap.Options) {
+			level := zap.NewAtomicLevelAt(zapcore.Level(-level))
+			options.Level = &level
+			options.Development = dev
+		})
+	} else {
+		// we need to create ZapLogger manually because we don't want to use Sampler, that does not support arbitrary log levels
+		encoder := zapcore.NewJSONEncoder(zap.NewProductionEncoderConfig())
+		sink := zapcore.AddSync(os.Stderr)
+		ZapLogger = zap.New(
```
Comment (nmiculinic) [inline pkg/util/cmd.go:50]: Add big warning somewhere we're setting this global variable here.

Comment (aborilov) [reply pkg/util/cmd.go:50]: I really don't like it, do you have any ideas on how to get rid of it?

Comment (aborilov) [reply pkg/util/cmd.go:50]: done

File: pkg/util/cmd_test.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestBuildLogger(t *testing.T) {
+	var buf bytes.Buffer
+	log := BuildLogger(7, false, &buf)
+	log.V(8).Info("hello")
+	assert.True(t, buf.Len() == 0)
```
Comment (nmiculinic) [inline pkg/util/cmd_test.go]:
```suggestion
	assert.Empty(t, buf)
```

File: pkg/util/cmd_test.go [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+	log.V(6).Info("hello")
+	assert.False(t, buf.Len() == 0)
```
Comment (nmiculinic) [inline pkg/util/cmd_test.go]:
```suggestion
	assert.NotEmpty(t, buf)
```

**Review by nmiculinic — CHANGES_REQUESTED** [review]
overall looks good, could you just use the right `assert.X` verb in the tests.

File: pkg/util/cmd_test.go [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+	buf.Reset()
+
+	log = BuildLogger(7, true, &buf)
+	log.V(8).Info("hello")
+	assert.True(t, buf.Len() == 0)
+	log.V(6).Info("hello")
+	assert.False(t, buf.Len() == 0)
+	err = json.Unmarshal(buf.Bytes(), &v)
+	assert.NotNil(t, err)
+	fmt.Println(buf.String())
```
Comment (nmiculinic) [inline pkg/util/cmd_test.go]: Also, don't use `Println` in the test, use `t.Log`
```suggestion
	t.Log(buf.String())
```

Comment (aborilov) [reply pkg/util/cmd_test.go]: meh, I forget to remove this line

Comment (aborilov) [reply pkg/util/cmd_test.go]: done

File: pkg/util/cmd_test.go (hunk 3 above)
Comment (aborilov) [reply pkg/util/cmd_test.go]: done

File: pkg/util/cmd_test.go (hunk 4 above)
Comment (aborilov) [reply pkg/util/cmd_test.go]: done

**Pull request merged**